model/model.go

```diff
@@ -764,6 +769,31 @@ func (pi *PartitionInfo) GetNameByID(id int64) string {
 	return ""
 }
 
+func (pi *PartitionInfo) GetStateByID(id int64) (SchemaState, bool) {
+	for _, pstate := range pi.States {
+		if pstate.ID == id {
+			return pstate.State, true
+		}
+	}
+	return StateNone, false
+}
+
+func (pi *PartitionInfo) SetStateByID(id int64, state SchemaState) bool {
+	for i, pstate := range pi.States {
+		if pstate.ID == id {
+			pstate.State = state
+			pi.States[i] = pstate
+			return true
+		}
+	}
+	return false
+}
```

Comment:
`States` is empty by default. If you set one of them to `StateGlobalTxnWriteOnly`, it always fails.

Reply:
I think we can init `states` in `tidb` repo while  `PartitionInfo` can provide some useful methods here.

Reply:
Alternatively, you can save the states that are not `StatePublic` in `pi.States`.
To this end, just append a new `PartitionState` if it's not found in `pi.States`.

Reply:
Is there any practical method to init `States`? Sometimes `TableInfo` is initialized by `json.Unmarshal`, and it's not elegant to fill `States` after unmarshalling.

Reply:
`To this end, just append a new PartitionState if it's not found in pi.States` I think this is a good idea.